src/Vlingo.Actors.Tests/ActorFactoryTest.cs

```diff
@@ -100,7 +100,7 @@ public void TestConstructorFailure()
                 address.Name);
             var mailbox = new TestMailbox();
 
-            Assert.Throws<ArgumentException>(() => ActorFactory.ActorFor(
+            Assert.Throws<ApplicationException>(() => ActorFactory.ActorFor(
```

Comment:
It was better before. If didn't see the rest of the code but if this is the MS `ApplicationException` so this is usless.
From: https://msdn.microsoft.com/en-us/library/System.ApplicationException

> You should derive custom exceptions from the Exception class rather than the ApplicationException class. You **should not throw an ApplicationException** exception in your code, and you should not catch an ApplicationException exception unless you intend to re-throw the original exception.

One simple reason is that there are other exception classes in .NET derived from `ApplicationException`. If you throw `ApplicationException` in your code and catch it later, you might also catch the derived exceptions which might break your application.

Reply:
Good point. However, the ArgumentException was wrongly used. It is raised in a different path which this test in question is not executing. Fixed that now.